src/utils.ts

```diff
@@ -50,16 +60,71 @@ export function assert(expression: boolean, message: string): void {
 export type Mutable<T> = { -readonly [P in keyof T]: T[P] };
 
 export function getValueWithOptions<DataType>(
-  value: unknown,
-  options?: GetOptions<DataType>
+  value: string,
+  options: Omit<GetOptions<DataType>, "serializer"> = {
+    deserializer: deserialize,
+  }
 ): DataType | null {
-  if (value === undefined || value === null) {
+  const deserialized = deserialize(value);
+
+  if (deserialized === undefined || deserialized === null) {
     return options?.defaultValue ?? null;
   }
 
   if (options?.reconstructor) {
-    return options.reconstructor(value);
+    return options.reconstructor(deserialized);
   }
 
-  return value as DataType;
+  return deserialized as DataType;
+}
+
+export function serializeValueWithOptions<DataType>(
+  value: DataType,
+  { serializer }: Pick<GetOptions<DataType>, "serializer"> = {
+    serializer: serialize,
+  }
+): string {
+  return serializer(value);
+}
+
+export function serialize(valueToSerialize: unknown): string {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  Date.prototype.toJSON = function () {
+    return {
+      value: this.toISOString(),
+      [TYPE_KEY]: TypeBrand.DATE,
+    };
+  };
+
+  return JSON.stringify(valueToSerialize, (_, value) => {
+    if (typeof value === "bigint") {
```

Comment:
I think it's better to check type isinstance of Date instead of override `Date.prototype`?

Reply:
Unfortunately, by the time you get the value argument inside the `stringify`replacer, the `toJSON` function of the `Date`object is already called so I cannot infer the type of the variable, I tried several ways to make it happen but this was the only one that worked, evem though I don't like it's hacky nature/playing around with prototypes

Reply:
I see, thanks for clarification!

Reply:
Also doesn't have luck with value - as you said it's already string, but I found this works: https://stackoverflow.com/a/54037861

Reply:
Thank you. I just updated the `Date` serialization function with this suggestion and it seems to work as expected.